src/ParseObject.js

```diff
@@ -2375,6 +2384,12 @@ const DefaultController = {
         if (el instanceof ParseFile) {
           filesSaved.push(el.save(options));
         } else if (el instanceof ParseObject) {
+          if (allowCustomObjectId && !el.id) {
```

Comment:
This forces all objects to have custom ids, if custom ids are allowed.
Is this intentional? This is not how the server treats allowCustomObjectId.

Reply:
I second this. Just turned this on, and this does not work as expected.

Reply:
I recommend looking at https://github.com/parse-community/Parse-SDK-JS/pull/1309#issuecomment-909436697